XRTK-Core/Packages/com.xrtk.core/Services/TeleportSystem/MixedRealityTeleportSystem.cs

```diff
@@ -213,12 +213,17 @@ private void ProcessTeleportationRequest(TeleportEventData eventData)
 
             var cameraTransform = MixedRealityToolkit.CameraSystem == null
                 ? CameraCache.Main.transform
-                : MixedRealityToolkit.CameraSystem.CameraRig.CameraTransform;
+                : MixedRealityToolkit.CameraSystem.MainCameraRig.CameraTransform;
             var cameraPosition = cameraTransform.position;
             var cameraParent = cameraTransform.parent;
-            Debug.Assert(cameraParent != null, "Teleport system requires that the camera be parented under another object.");
+            Debug.Assert(cameraParent != null, "The Teleport System requires that the camera be parented under another object.");
             var height = targetPosition.y;
 
+            if (MixedRealityToolkit.CameraSystem != null)
+            {
+                height += MixedRealityToolkit.CameraSystem.MainCameraRig.PlayspaceTransform.position.y;
```

Comment:
We can probably just use the cameraTransform.parent.position.y so we don't have to check for the camera system again.